### ISSUE TITLE
Two overlapped transparent boxshapes， sometimes innerbox disappear

### ISSUE THREAD

**Issue by jiangwenming2021** [o3de/o3de#5971]
**Describe the bug**
Two overlapped transparent boxshapes， sometimes innerbox disappear.

**Steps to reproduce**
1. add two entities in the scene editor.
2. add boxshape component to each other with default settings.
3. move them to (0,0,0)
4. modify one entity's uniform scale to 2, and name it to outbox
5. name the other to inner box
6. now we can see both outbox and inner box as shown Fig1.
7. just move inner box a bit, e.g. (0,0,0.04), then the inner box disappeared as shown in Fig2.

**Expected behavior**
two box can be seen whether they are overlapped.

**Actual behavior**
sometimes the inner box disappeared.

**Assets required**
NA

**Screenshots/Video**
Fig 1:
![image](https://user-images.githubusercontent.com/80514815/143676560-f5f914e5-641a-41c6-9eb8-62346aa67f28.png)

Fig 2:
![image](https://user-images.githubusercontent.com/80514815/143676567-c2b33d89-ca1e-4a03-8bfd-2357cfb7fa0a.png)

**Found in Branch**
Name of or link to the branch where the issue occurs.

**Desktop/Device (please complete the following information):**
 - Device: [e.g. PC] 
 - OS: [e.g. Windows]
 - Version [e.g. 10, Bug
 - CPU [e.g. Intel I7 ]
 - GPU [NVidia RTX 5000]
 - Memory [e.g. 16GB]

**Comment by hultonha** [comment]
I think this is more likely a rendering issue. I may need to discuss with @rgba16f next year to find a way to handle this.

**Comment by rgba16f** [comment]
Yes this is a common transparency rendering issue when transparent objects write to the depth buffer and overlap each other. Transparent objects are notorious for having order of rendering dependencies. 

Pinging @invertednormal who has spent time studying some of the various full solutions to transparent objects (aka Order Independent Transparency). 

The solution that gets picked here depends on the need for correctness and performance. If some rendering issues are acceptable then the simplest solution is often to turn off depth writes for transparent objects (they will still test against the depth buffer for their depth clipping). 

Beyond that there are various techniques that can give either a plausible or perfectly correct result per pixel but that are expensive on the gpu.

Dealing with transparency is a deep topic :)

**Comment by invertednormal** [comment]
I don't think there's any ordering with how aux geom (like box shapes) are rendered (correct me if I'm wrong @rgba16f ), but transparent meshes are already sorted back-to-front, and there's a sort key in the mesh component that can be used to override sort order. This can be used to fix most transparency issues with actual game meshes for very little cost. Of course this can't solve all transparency ordering problems, and there isn't currently any order independent transparency technique in o3de.    There are several ways to go about that, but they can add a fair amount of cost to rendering.

**Comment by hultonha** [comment]
Thanks very much for the detailed response @rgba16f and @invertednormal... I wonder if this is something we can fix on the client side (in terms of the shapes using the DebugDraw/AtomAuxGeom API). Perhaps if we update those to not write to the depth buffer, things might not look perfect but hopefully better than they do today? This is potentially something @jonawals could look at when they're back along with the Viewport Visual Feedback work. Thanks again!
